### PR TITLE
remove GTimeVal following glib 2.61.2 deprecation

### LINKED ISSUE #1144: warning: ‘GTimeVal’ is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
Hi,

This pops up several times when building irssi with a recent glib. Maybe this should be looked at.

Relevant: https://tecnocode.co.uk/2019/08/24/gtimeval-deprecation-in-glib-2-61-2/

### PR DESCRIPTION
**Pull request by ailin-nemui**
fixes #1144

### CODE REVIEW

**Comment by ailin-nemui** [comment]
@dequis #1141

**Comment by dequis** [comment]
```
15:36 < dx> [1:bitlbee (change with ^X)] [Lag: 973.33]
15:36 < dx> lol i guess
15:36 < dx> i started irssi before bitlbee, and that's the time since it started the reconnection attempts, shows up as lag for a brief moment
15:39 < dx> (it does this for every network)
```

Restarted today with git master + this PR merged in.

**Comment by ailin-nemui** [comment]
are you sure that this only happens with this patch?

**Comment by dequis** [comment]
Not really, i just commented to not forget about it and check later.

**Comment by dequis** [comment]
Confirming: with this commit, irssi shows [Lag: 88.42] for one network, weirdly always that number, a few seconds after startup / successfully connecting to a network. Not correlated to reconnection time as I previously thought.

Valgrind doesn't whine about uninitialized reads but I guess this one doesn't hit a condition.

File: src/irc/core/lag.c [hunk 1]
```diff
@@ -48,18 +48,18 @@ static void lag_ping_error(IRC_SERVER_REC *server)
...
+	if (server->lag_sent == 0) {
 		/* not expecting lag reply.. */
 		return;
 	}
 
-	g_get_current_time(&now);
-	server->lag = (int) get_timeval_diff(&now, &server->lag_sent);
-	memset(&server->lag_sent, 0, sizeof(server->lag_sent));
+	now = g_get_real_time();
+	server->lag = now - server->lag_sent;
```
Comment (dequis) [inline src/irc/core/lag.c]:
```suggestion
	server->lag = (now - server->lag_sent) / G_TIME_SPAN_MILLISECOND;
```

Found the issue! It wasn't some uninitialized nonsense, it was storing microseconds in field that expects milliseconds!

Comment (ailin-nemui) [reply src/irc/core/lag.c]: thanks!

**Comment by ailin-nemui** [comment]
@irssi/developers

**Pull request merged**